Fix segfaults in synctex binary
Patch by Sebastian Ramacher <sramacher@debian.org>
Forwarded upstream
---
 texk/web2c/synctexdir/synctex_main.c |   13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

--- texlive-bin.orig/texk/web2c/synctexdir/synctex_main.c
+++ texlive-bin/texk/web2c/synctexdir/synctex_main.c
@@ -304,11 +304,11 @@
 				if(++arg_index<argc) {
 					goto option_command;
 				} else {
-					synctex_view_proceed(&Ps);
+					return synctex_view_proceed(&Ps);
 				}
 			} else {
 				Ps.directory = getenv("SYNCTEX_BUILD_DIRECTORY");
-				synctex_view_proceed(&Ps);
+				return synctex_view_proceed(&Ps);
 			}
 		}
 option_command:
@@ -320,7 +320,7 @@
 					if(++arg_index<argc) {
 						goto option_hint;
 					} else {
-						synctex_view_proceed(&Ps);
+						return synctex_view_proceed(&Ps);
 					}
 				} else {
 					/* retrieve the environment variable */
@@ -329,7 +329,7 @@
 				}
 			} else {
 				Ps.viewer = getenv("SYNCTEX_VIEWER");
-				synctex_view_proceed(&Ps);
+				return synctex_view_proceed(&Ps);
 			}
 		}
 option_hint:
@@ -347,7 +347,7 @@
 						*Ps.after = '\0';
 						if(Ps.offset<strlen(Ps.middle)) {
 							++Ps.after;
-							synctex_view_proceed(&Ps);
+							return synctex_view_proceed(&Ps);
 						}
 					}
 				}
@@ -460,11 +460,12 @@
 					break;
 				}
 				/* copy the rest of viewer into the buffer */
-				if(buffer_cur != memcpy(buffer_cur,viewer,strlen(viewer))) {
+				if(buffer_cur != strncpy(buffer_cur,viewer,size + 1)) {
 					synctex_help_view("Memory copy problem");
 					free(buffer);
 					return -1;
 				}
+				buffer_cur[size] = '\0';
 				printf("SyncTeX: Executing\n%s\n",buffer);
 				status = system(buffer);
 				free(buffer);
